framework/base/Theme.php

```diff
@@ -99,7 +99,13 @@ public function getBaseUrl()
      */
     public function setBaseUrl($url)
     {
-        $this->_baseUrl = rtrim(Yii::getAlias($url), '/');
+        $baseUrl = null;
+        
+        if ($url !== null) {
+            $baseUrl = rtrim(Yii::getAlias($url), '/');
+        }
+
+        $this->_baseUrl = $baseUrl;
```

Comment:
these 6 lines should be compressed to one line as it will still be understandable.